cmd/api/database_test.go

```diff
@@ -234,23 +238,40 @@ func TestListBooks(t *testing.T) {
 		testBookslist = append(testBookslist, b)
 	}
 
-	t.Run("List all books, no filtering, without errors", func(t *testing.T) {
+	t.Run("List all books, no filtering, without errors.", func(t *testing.T) {
 		is := is.New(t)
 
-		//Asking all books on the list
-		returnedBooks, err := listBooks("", 0.00, 9999.99, "name", "asc", true)
+		//Asking all books on the list. Expected 30 books on page 1.
+		itemsTotal, err := countRows("", 0.00, 9999.99, true)
+		is.NoErr(err)
+		is.True(itemsTotal == 30)
+		returnedBooks, err := listBooks("", 0.00, 9999.99, "name", "asc", true, 30, 0)
 		is.NoErr(err)
 		for i, expected := range testBookslist {
 			compareBooks(is, returnedBooks[i], expected)
 		}
 	})
 
+	t.Run("List books with limited page size, without errors.", func(t *testing.T) {
+		is := is.New(t)
+
+		//Asking second 10 books of the list. With page_size=10, it would be page=2;
+		itemsTotal, err := countRows("", 0.00, 9999.99, true)
+		is.NoErr(err)
+		is.True(itemsTotal == 30)
+		returnedBooks, err := listBooks("", 0.00, 9999.99, "name", "asc", true, 10, 10)
+		is.NoErr(err)
+		for i, expected := range testBookslist[10:19] {
+			compareBooks(is, returnedBooks[i], expected)
+		}
+	})
```

Comment:
This is good, but please write a test going over all the pages and making sure we are returning all the books.

Reply:
Added a trello task for that.